VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -106,32 +108,57 @@ public void onPageScrollStateChanged(int state) {
         mPreviousBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                int index = pager.getCurrentItem();
-                if (index > 0) {
-                    PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
-                    if (curView != null)
-                        curView.onPageDisappear();
-                    pager.setCurrentItem(index - 1);
-                }
+                navigatePreviousComponent();
             }
         });
         mNextBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                int index = pager.getCurrentItem();
-                if (index < pagerAdapter.getCount() - 1) {
-                    PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
-                    if (curView != null)
-                        curView.onPageDisappear();
-                    pager.setCurrentItem(index + 1);
-                }
+                navigateNextComponent();
             }
         });
+    }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
         updateUIForOrientation();
     }
 
     @Override
+    public boolean navigatePreviousComponent() {
+        boolean ret = false;
+
+        int index = pager.getCurrentItem();
+        if (index > 0) {
+            PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
+            if (curView != null)
+                curView.onPageDisappear();
+            pager.setCurrentItem(index - 1);
+            ret = true;
+        }
+
```

Comment:
Could you remove all of these empty lines in this method and below (navigateNextComponent)